client/state/data-layer/wpcom/activity-log/delete-credentials/index.js

```diff
@@ -0,0 +1,64 @@
+/**
+ * External dependencies
+ *
+ * @format
+ */
+
+import i18n from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import { http } from 'state/data-layer/wpcom-http/actions';
+import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
+import { JETPACK_CREDENTIALS_DELETE, JETPACK_CREDENTIALS_STORE } from 'state/action-types';
+import { successNotice, errorNotice } from 'state/notices/actions';
+
+export const request = ( { dispatch }, action ) => {
+	const notice = successNotice( i18n.translate( 'Deleting credentials…' ), { duration: 4000 } );
+	const { notice: { noticeId } } = notice;
+
+	dispatch( notice );
+
+	dispatch(
+		http(
+			{
+				apiVersion: '1.1',
+				method: 'POST',
+				path: `/activity-log/${ action.siteId }/delete-credentials`,
+				body: { role: action.role },
+			},
+			{ ...action, noticeId }
+		)
+	);
+};
+
+export const success = ( { dispatch }, action ) => {
+	dispatch( {
+		type: JETPACK_CREDENTIALS_STORE,
+		credentials: {
+			main: null,
+		},
+		siteId: action.siteId,
+	} );
```

Comment:
If I were to guess what kind of action would feed into the reducers then I would expect a `_DELETE` action to eliminate the credentials from the state tree, not necessarily a `_STORE`.

In fact, we have the opportunity here in the action handler to just send out the HTTP call to delete the credentials while letting the reducers do the actual removal on this `_DELETE` action. If the network request fails then we can add the credentials back in with a `_STORE` generated from the original action.

At the time I don't see us actually removing the credentials with any network calls.

Reply:
I agree. Do you think this is a blocker for this now, or is it something we can think about optimizing when we retrofit with the state-machine design?